core/sdk-core/src/main/java/software/amazon/awssdk/core/endpointdiscovery/EndpointDiscoveryRefreshCache.java

```diff
@@ -29,6 +30,8 @@ public final class EndpointDiscoveryRefreshCache {
 
     private final EndpointDiscoveryCacheLoader client;
 
+    private String key;
```

Comment:
Is there any reason we need this field? It doesn't seem to be thread safe

Reply:
This is so that we can reference this field from within the .handle method defined for the future

Reply:
Can we derive the key in the handle method? Making this field thread safe is more error prone and may impact performance.

Reply:
Yeah, thats what Im going to do.